scripts/run_all_benchmarks.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+set -e
+
+# Runs all benchmarks for all pallets, for the Substrate node.
+# Should be run on a reference machine to gain accurate benchmarks
+# current reference machine: https://github.com/paritytech/substrate/pull/5848
+
+echo "[+] Running all benchmarks for Substrate"
+
+if [ $1 != "skip-build" ]
+then
+  cargo +nightly build --profile production --locked --features=runtime-benchmarks
+fi
+
+./target/production/substrate benchmark pallet \
+    --chain=dev \
+    --list |\
+  tail -n+2 |\
+  cut -d',' -f1 |\
+  uniq > "substrate_pallets"
+
+# For each pallet found in the previous command, run benches on each function
+while read -r line; do
+  pallet="$(echo "$line" | cut -d' ' -f1)";
+  folder_name="$(echo "${pallet#*_}" | tr '_' '-')";
+  echo "Pallet: $pallet, Folder Name: $folder_name";
+  # '!' has the side effect of bypassing errexit / set -e
+  ! ./target/production/substrate benchmark pallet \
+    --chain=dev \
+    --steps=50 \
+    --repeat=20 \
+    --pallet="$pallet" \
+    --extrinsic="*" \
+    --execution=wasm \
+    --wasm-execution=compiled \
+    --template=./.maintain/frame-weight-template.hbs \
+    --output="./frame/${folder_name}/src/weights.rs"
+done < "substrate_pallets"
+rm "substrate_pallets"
+
+# Benchmark base weights
+! ./target/production/substrate benchmark overhead \
+  --chain=dev \
+  --execution=wasm \
+  --wasm-execution=compiled \
+  --weight-path="./frame/support/src/weights/" \
+  --warmup=10 \
+  --repeat=100
+
+
+# This true makes sure that $? is 0 instead of
```

Comment:
Default exit code should be 0 when nothing failed?

Reply:
I changed it to fail if something goes wrong.

Reply:
I don't think we want the whole process to fail if something goes wrong. Want to try and do things best effort.

Reply:
But otherwise we will have unnoticed failures. In which case would it be an advantage to ignore an error here?

Reply:
yes, that would be best. some kind of error output file, but it is annoying for the process to quit in the middle of running all benchmarks.

Imagine you start the script and leave it running over night, expecting everything to be done, and instead it stops on the 3rd pallet. You could at least get all the other pallets benchmarked, and come back to fix the broken pallet.